field/R3BGladFieldMap.cxx

```diff
@@ -74,39 +75,73 @@ R3BGladFieldMap::R3BGladFieldMap(const char* mapName, const char* fileType)
     fYAngle = -14.;
     fZAngle = 0.;
 
-    fXmin = fYmin = fZmin = 0.;
-    fXmax = fYmax = fZmax = 0.;
-    fXstep = fYstep = fZstep = 0.;
-    fNx = fNy = fNz = 0;
+    fXmin = 0;
+    fYmin = 0;
+    fZmin = 0.;
+    fXmax = 0;
+    fYmax = 0;
+    fZmax = 0.;
+    fXstep = 0;
+    fYstep = 0;
+    fZstep = 0.;
+    fNx = 0;
+    fNy = 0;
+    fNz = 0;
     fScale = 1.;
-    fBx = fBy = fBz = NULL;
+    fBx = nullptr;
+    fBy = nullptr;
+    fBz = nullptr;
     fName = mapName;
     TString dir = getenv("VMCWORKDIR");
     fFileName = dir + "/field/magField/R3B/" + mapName;
     if (fileType[0] == 'R')
+    {
         fFileName += ".root";
+    }
     else
+    {
         fFileName += ".dat";
+    }
 
     fType = 2;
 }
 
+void R3BGladFieldMap::SetFieldfromCurrent(double current)
+{
+    R3BLOG_IF(fatal, TMath::Abs(current) > 3583.81, "GLAD current cannot be larger than 3583.81 A.");
+    fScale = current / 3583.81;
```

Comment:
fScale should be a negative number

Reply:
Yes, the current must be negative. For FAIR it will be negative and positive depending on the configuration. 
What is the right convention?
I am assuming a positive current for the standard FAIR configuration and negative one for the current setup.

Reply:
Is this ok or should we change the sign?

Reply:
For the root macros one has to call the R3BGladFieldMap class with these lines:

R3BGladFieldMap *magField = new R3BGladFieldMap("R3BGladMap");
magField->SetFieldfromCurrent(fMeasCurrent);

where fMeasCurrent corresponds to the measured GLAD current. For the Phase-0 setups this must be negative and in Amperes.